lib/de.ml

```diff
@@ -1382,6 +1382,306 @@ module Inf = struct
+          ( _fill_byte d
+          ; fill_bits lit d n)
+      in
+      let exception End in
+      let exception Invalid_distance in
+      let exception Invalid_distance_code in
+      let lit_mask = lit.Lookup.m in
+      let dist_mask = dist.Lookup.m in
+      try
+        let rec length () =
+          fill_bits lit d lit.Lookup.l ;
+          let value = lit.Lookup.t.(d.hold land lit_mask) land Lookup.mask in
+          let len = lit.Lookup.t.(d.hold land lit_mask) lsr 15 in
+          d.hold <- d.hold lsr len ;
+          d.bits <- d.bits - len ;
+          if value < 256
+          then ( unsafe_set_uint8 d.o d.o_pos value
+                ; WInf.add d.w value
+                ; d.o_pos <- d.o_pos + 1
+                ; length ())
+          else if value == 256 then raise_notrace End
+          else ( d.l <- value - 257; extra_length ())
+        and extra_length () =
+          let len = _extra_lbits.(d.l) in
+          fill_bits lit d len ;
+          let extra = pop_bits d len in
+          d.l <- _base_length.(d.l land 0x1f) + 3 + extra ;
+          distance ()
+        and distance () =
+          fill_bits lit d dist.Lookup.l ;
+          let value = dist.Lookup.t.(d.hold land dist_mask) land Lookup.mask in
+          let len = dist.Lookup.t.(d.hold land dist_mask) lsr 15 in
+          d.hold <- d.hold lsr len ;
+          d.bits <- d.bits - len ;
+          d.d <- value ;
+          extra_distance ();
+        and extra_distance () =
+          let len = _extra_dbits.(d.d land 0x1f) in
+          fill_bits lit d len ;
+          let extra = pop_bits d len in
+          d.d <- _base_dist.(d.d) + 1 + extra ;
+          write ()
+        and write () =
+          if d.d == 0 then raise_notrace Invalid_distance_code ;
+          if d.d > WInf.have d.w then raise_notrace Invalid_distance ;
+          let len = min d.l (bigstring_length d.o - d.o_pos) in
+          let off = WInf.mask (d.w.WInf.w - d.d) in
+          if d.d == 1
+          then
+            ( let v = unsafe_get_uint8 d.w.WInf.raw off in
+              WInf.fill d.w v d.o d.o_pos len )
+          else
+            ( let off = WInf.mask (d.w.WInf.w - d.d) in
+              let pre = WInf.max - off in
+              let rst = len - pre in
+              if rst > 0
+              then ( WInf.blit d.w d.w.WInf.raw off d.o d.o_pos pre
+                    ; WInf.blit d.w d.w.WInf.raw 0 d.o (d.o_pos + pre) rst )
+              else WInf.blit d.w d.w.WInf.raw off d.o d.o_pos len ) ;
+          d.o_pos <- d.o_pos + len ;
+          if d.l - len == 0 then length () else d.l <- d.l - len
+        in
+        length ()
+      with End -> ()
+          | Invalid_distance -> err_invalid_distance ()
+          | Invalid_distance_code -> err_invalid_distance_code ()
+
+    let fixed d =
+      inflate fixed_lit fixed_dist d
+
+    (* XXX(clecat): The table functions are almost a copy of the stream implementation, by
+       adapting their code, they should be easily merged *)
```

Comment:
Please do.

Reply:
I'll push the other modifications for now and change that in a separate commit, so we keep clear tracks of it

Reply:
Seems like good enough, this part is not the most important to review.